Tests/SwiftKafkaTests/KafkaProducerTests.swift

```diff
@@ -0,0 +1,102 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@testable import SwiftKafka
+import XCTest
+
+final class KafkaProducerTests: XCTestCase {
```

Comment:
Can we add an assert that after calling `shutdownGracefully` that the `KafkaProducer` is deinited. You can do this by creating a weak ref before then shutting down and dropping the strong ref. Afterwards, you can assert that the weak ref is `nil`.